### PR TITLE
[WIP] add geometries HTML section, add Leaflet draw controls

### PR DESCRIPTION
**Pull request by nvkelso**
Changes:

- [x] Add new Leaflet Draw controls on right side of map
    - [x] Contextually adds circleMarker (label centroid and revGeo centroid), polygon (default geom), and rectangle (label bbox) options if those don't yet exist
    - [x] Prompt for which centroid geoms to add when adding a circlemarker
    - [x] Add ability to reset the edited geometries
    - [x] Draw new geoms using right colour
    - [ ] Not fully wired up to ~~create~~, edit, ~~delete~~ WOF geometries (it's unlinked now) in Leaflet Draw layer group
    - [ ] Not fully wired up to take those layer group edits and update the WOF record with them
    - [ ] v1 probably won't allow editing polygon geoms that aren't mapzen / wof sourced? Otherwise have to deal with alt geoms, oh my!?
- [x] Add new `Geometries` section at very top near map
    - [x] Checked is true for following if that type of geom already exists (as it can be hard to see on the map). Disabled (not check or uncheckable) for now – maybe in future it could delete or add that geom type?
    - [x]  Default geom (geometry) with source and area
    - [x]  Alternate geoms (geometry) and count of sources
    - [x]  Math centroid (geom.longitude, geom.latitude)
    - [x]  Label centroid (lbl.longitude, lbl.latitude)
    - [x]  Label bounding box (lbl.bbox)
    - [x]  Reverse geocoding centroid (reversegeo.longitude, reversegeo.latitude)
- [x] Add new header for existing `Common properties` section
- [ ] Adjust map geom colors (geom centroid should be black?)

Testing locations:

- [San Francisco](http://127.0.0.1:5000/place/edit?url=https://raw.githubusercontent.com/whosonfirst-data/whosonfirst-data-admin-us/master/data/859/225/83/85922583.geojson), CA, USA - `85922583` - Polygon, has label bbox and centroid, no revGeo centroid, multiple alternate geometries
- [New York](http://127.0.0.1:5000/place/edit?url=https://raw.githubusercontent.com/whosonfirst-data/whosonfirst-data-admin-us/master/data/859/775/39/85977539.geojson), NY, USA - `85977539` - Polygon, no label bbox, has label centroid, has revGeo centroid, multiple alternate geometries
- [Olopoto](http://127.0.0.1:5000/place/edit?url=https://raw.githubusercontent.com/whosonfirst-data/whosonfirst-data-admin-ng/master/data/132/729/337/1/1327293371.geojson), Nigera - `1327293371`  - Point, no extras at all. Now zooms out to the label bbox on init map / page load, no alternate geometries

Screenshot:

image

### CODE REVIEW

File: editor/templates/place/edit.html [hunk 1]
```diff
@@ -5,6 +5,7 @@
 {%- block styles %}
 {{ super() }}
 <link rel="stylesheet" href="https://unpkg.com/leaflet@1.3.1/dist/leaflet.css" integrity="sha512-Rksm5RenBEKSKFjgI3a41vrjkw4EVPlJ3+OiI65vTjIdo9brlAacEuKOiQ5OFh7cOI1bkDwLqdLw3Zg0cRJAAQ==" crossorigin=""/>
+<link rel="stylesheet" href="https://unpkg.com/leaflet-draw/dist/leaflet.draw.css" />
```
Comment (nvkelso) [inline editor/templates/place/edit.html]: Should this have an integrity check like main Leaflet import?

File: editor/templates/place/edit.html [hunk 2]
```diff
@@ -19,6 +20,7 @@
 {{ super() }}
 <script src="https://unpkg.com/leaflet@1.3.1/dist/leaflet.js" integrity="sha512-/Nsx9X4HebavoBvEBuyp3I7od5tA0UzAxs+j83KgC8PU0kgB4XiK4Lfe4y4cgBtaRJQEIFCW+oC506aPT2L1zw==" crossorigin=""></script>
 <script src="https://unpkg.com/leaflet-ajax@2.1.0/dist/leaflet.ajax.min.js" integrity="sha384-sKs8ZrrxyJoElcPVznZwGpUTTXvkMYfHYxdIFzO8Hd0TA6emONMj8BwnsFf+6cZ/" crossorigin=""></script>
+<script src="https://unpkg.com/leaflet-draw/dist/leaflet.draw.js"></script>
```
Comment (nvkelso) [inline editor/templates/place/edit.html]: Should this have an integrity check like main Leaflet import? and crossorigin set?

File: editor/templates/place/edit.html [hunk 3]
```diff
@@ -90,19 +106,53 @@
         L.latLng({{ bboxParts[1] }}, {{ bboxParts[0] }}),
         L.latLng({{ bboxParts[3] }}, {{ bboxParts[2] }}),
     );
-    var labelBboxLayer = L.rectangle(labelBbox, {color: geomColors.labelBbox}).addTo(map);
+    var labelBboxLayer = L.rectangle(labelBbox, {color: geomColors.labelBbox, title: 'label bounding box'}).addTo(drawnItems);
+    map.fitBounds(featureLayer.getBounds().extend(labelBboxLayer.getBounds()));
+    // Point default geoms sometimes have a label bounding box that is larger, zoom out to it
```
Comment (nvkelso) [inline editor/templates/place/edit.html]: (This is a bug fix)

File: editor/templates/place/edit.html [hunk 4]
```diff
@@ -212,6 +262,64 @@ <h3>Editing {{ wof_doc.properties['wof:name'] or wof_doc.id }}</h3><small>(<a hr
 </div>
 
 <form class="needs-validation" novalidate action="{{ url_for('place.edit_place', url=wof_url) }}" method="POST">
+    <h4>Geometries</h4>
+    <!-- https://github.com/whosonfirst/whosonfirst-properties/blob/master/properties/wof/name.json -->
+    <div class="form-group">
+        {%- if wof_doc.properties['geom:area_square_m'] is defined %}
+        <small id="wof-name-help" class="form-text text-muted">Land-only multi-polygons are preferred, points are allowed. {{ wof_doc.properties['geom:area_square_m'] }} square meters</small>
```
Comment (nvkelso) [inline editor/templates/place/edit.html]: The area here should be rounded to an int

File: editor/templates/place/edit.html (hunk 1 above)
Comment (iandees) [reply editor/templates/place/edit.html]: Yep. If you add `?meta` to the unpkg URL, it'll give you the integrity check string to add:

```
sha384-NZLkVuBRMEeB4VeZz27WwTRvlhec30biQ8Xx7zG7JJnkvEKRg5qi6BNbEXo9ydwv
```

File: editor/templates/place/edit.html [hunk 5]
```diff
@@ -212,6 +262,64 @@ <h3>Editing {{ wof_doc.properties['wof:name'] or wof_doc.id }}</h3><small>(<a hr
...
+        {%- endif %}
+        <input type="checkbox" id="geom_default" name="geom_default" value="Default geom" disabled=True checked=True>
+        <label for="geom_default">Default geometry</label>&nbsp;<small>(<code>geometry</code>)</small>
+        {%- if wof_doc.properties['src:geom_alt'] is defined %}
+        <small id="wof-name-help" class="form-text text-muted">Alternate geometries are present from the following {{ wof_doc.properties['src:geom_alt']|length }} sources.</small>
+        <input type="checkbox" id="geom_alternates" name="geom_alternates" value="Alternate geometries" disabled=True checked=True>
+        <label for="geom_default">Alternate geometries</label>&nbsp;<small>(<code>src:geom_alt</code>)</small>
+        {%- for alt_geom in wof_doc.properties['src:geom_alt'] %}
+        <ul class="form-text text-muted">
+            <li class="col-md-2 col-form-label">{{ alt_geom }}</li>
```
Comment (nvkelso) [inline editor/templates/place/edit.html]: The CSS styling here is too big in the list. Ideally it's "small", or instead of the list it's a series of inline text chunks at the end of the label line above (with special handling for commas & etc).

File: editor/templates/place/edit.html (hunk 4 above)
Comment (nvkelso) [inline editor/templates/place/edit.html]: To match the sequencing in the Common properties section of the form, the small helper text should appear _after_ the input (not above)... but the CSS would need a variant added else there's too much spacing added _above_ making it look like the text is for the next input not the current input.

(Applies to all in the geometries section)

File: editor/templates/place/edit.html (hunk 1 above)
Comment (nvkelso) [reply editor/templates/place/edit.html]: Thanks! 

Resolved via https://github.com/iandees/wof-editor/pull/85/commits/65cd7b0ed6213e4747b1a66916d62f42c7d8cd36.

File: editor/templates/place/edit.html (hunk 2 above)
Comment (nvkelso) [reply editor/templates/place/edit.html]: Resolved via https://github.com/iandees/wof-editor/pull/85/commits/65cd7b0ed6213e4747b1a66916d62f42c7d8cd36.
